Source/WebCore/page/text-extraction/TextExtraction.cpp

```diff
@@ -1255,5 +1256,208 @@ void handleInteraction(Interaction&& interaction, Page& page, CompletionHandler<
     completion(false, "Invalid action"_s);
 }
 
+static constexpr auto maxDescriptionLength = 512;
+
+static String normalizeText(const String& string)
+{
+    auto result = makeStringByReplacingAll(string, '\'', ""_s);
+    result = makeStringByReplacingAll(string, '\"', ""_s);
+    result = makeStringByReplacingAll(result, '\n', " "_s);
+    result = makeStringByReplacingAll(result, '\r', ""_s);
+    result = result.trim(isASCIIWhitespace<char16_t>);
+    if (result.length() <= maxDescriptionLength)
+        return result;
+
+    return makeString(result.left(maxDescriptionLength / 2 - 2), "..."_s, result.right(maxDescriptionLength / 2 - 1));
+}
+
+static String normalizedLabelText(const Element& element)
+{
+    for (auto attribute : std::array { HTMLNames::aria_labelAttr.get(), HTMLNames::labelAttr.get() }) {
```

Comment:
This can probably be an initializer list
```suggestion
    for (auto attribute : { HTMLNames::aria_labelAttr.get(), HTMLNames::labelAttr.get() }) {
```